fix(core): assign labels as aria-labelledby for corresponding controls inside fd-form-item

dmitry-stepanenko opened the pull request:
## Related Issue(s)

<!-- If this PR fixes multiple issues, please use the full syntax(`closes #issue-number`) for each issue so that each issue gets automatically closed on PR merge; for example: `closes #0001, closes #0002`, and so on. -->

closes none

## Description
Adding logic, that ties labels with controls inside `fd-form-item` block
 ```html
<div fd-form-item>
  <label fd-form-label for="input-id">Label Text</label>
  <fd-select></fd-select> <= should have "aria-labelledby" property pointing to the corresponding label
</div>
  ```
<!-- Enter short description of the change -->

## Screenshots

<!-- If you've made any style changes, please provide appropriate screenshots (before and after) to help reviewers. -->

### Before:

### After:

#### Please check whether the PR fulfills the following requirements

##### During Implementation

1. Visual Testing:

-   [n/a] visual misalignments/updates
-   [n/a] check Light/Dark/HCB/HCW themes
-   [n/a] RTL/LTR - proper rendering and labeling
-   [n/a] responsiveness(resize)
-   [n/a] Content Density (Cozy/Compact/(Condensed))
-   [n/a] States - hover/disabled/focused/active/on click/selected/selected hover/press state
-   [n/a] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
-   [n/a] Mouse vs. Keyboard support
-   [n/a] Text Truncation

2. API and functional correctness

-   [x] check for console logs (warnings, errors)
-   [x] API boundary values
-   [n/a] different combinations of components - free style
-   [x] change the API values during testing

3. Documentation and Example validations

-   [x] missing API documentation or it is not understandable
-   [n/a] poor examples
-   [n/a] Stackblitz works for all examples

4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality

-   [x] the commit message(s) follows the guideline:
        https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
-   [n/a] tests for the changes that have been done
-   [x] all items on the PR Review Checklist are addressed :
        https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
-   [n/a] Run npm run build-pack-library and test in external application
-   [n/a] update `README.md`
-   [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

InnaAtanasova commented:
<img width="1301" alt="Screen Shot 2022-06-01 at 1 42 43 PM" src="https://user-images.githubusercontent.com/39598672/171468989-a38b165b-50ac-4f2c-9048-f6ec06b7069a.png">

Hi @dmitry-stepanenko , could you briefly explain what you fixed with this PR? What is the issue you are trying to fix? Why do every list item need `aria-describedby`? I see an empty hidden span added. What's it's purpose when there's no label or content? Lastly, since this is Core, could you use IDs that start with `fd-`?

dmitry-stepanenko commented:
> <img alt="Screen Shot 2022-06-01 at 1 42 43 PM" width="1301" src="https://user-images.githubusercontent.com/39598672/171468989-a38b165b-50ac-4f2c-9048-f6ec06b7069a.png">
> 
> Hi @dmitry-stepanenko , could you briefly explain what you fixed with this PR? What is the issue you are trying to fix? Why do every list item need `aria-describedby`? I see an empty hidden span added. What's it's purpose when there's no label or content? Lastly, since this is Core, could you use IDs that start with `fd-`?

Hi @InnaAtanasova. Your points are not related to this PR, because I'm not touching list options, only tying actual inputs with their labels for screen readers. This was raised as part of #5574 (issue №8 there), but I identified this as a common problem across the UI library.
<img width="1349" alt="image" src="https://user-images.githubusercontent.com/33101123/171483231-1bcb701a-6a26-405f-b219-15d6c5a82a06.png">


But you're right, it's not really reasonable to show empty element there, added `*ngIf`. And changed id's prefix to be `fd` instead

dmitry-stepanenko commented:
> Should we clean up in the docs all the components that have the ariaLabelledBy set manually where it doesn't match with anything? For example here: https://github.com/SAP/fundamental-ngx/blob/main/apps/docs/src/app/core/component-docs/combobox/examples/combobox-example.component.html#L7

@mikerodonnell89 good point, done


Had to create a new PR as this one cannot be reopened https://github.com/SAP/fundamental-ngx/pull/8223
